src/App.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import Header from './components/Header';
+import Editor from './components/Editor';
+import Canvas from './components/Canvas';
+import './styles/App.scss';
+import {TInfographicsSection} from './types/TInterfaceSection';
+
+interface IState {
+    infographicsSections: TInfographicsSection[],
+    numberOfAddedSections: number,
+    selectedSectionId: number | null
+}
+
+class App extends React.Component<{}, IState> {
+
+    state = {
+        infographicsSections: [] as TInfographicsSection[],
+        numberOfAddedSections: 0,
+        selectedSectionId: null
+    };
+
+    public render() {
+
+        const {infographicsSections, selectedSectionId} = this.state;
+
+        return (
+            <div className="app">
+                <Header/>
+                <div className="app-wrapper">
+                    <Editor
+                        infographicsSections={infographicsSections}
+                        selectedSectionId={selectedSectionId}
+                        addSection={this.handleAddSection}
+                        deleteSection={this.handleDeleteSection}
+                    />
+                    <Canvas
+                        infographicsSections={infographicsSections}
+                        selectedSectionId={selectedSectionId}
+                        handleSelectSection={this.handleSelectSection}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private handleAddSection = () => {
+
+        const {infographicsSections, numberOfAddedSections, selectedSectionId} = this.state;
+
+        const newSection: TInfographicsSection = {
+            id: numberOfAddedSections,
```

Comment:
pokud to chces delat tak bych nejdriv zvysil cislo, to priradil a pak ulozil do state, protoze takhle mas vzdy o 1 cislo vyssi nez kolik tam je realne

zvazil bych `numberOfAddedSections` hodit jako lokalni variable, protoze do state musis davat veci jen pokud potrebujes renderovat znovu komponentu

takze pod state si udelas

`numberOfAddedSections: number = 0`

 a tady pak budes jen delat
`this.numberOfAddedSections += 1`
`id: this.numberOfAddedSections`